src/handlers/permissions-revoke.ts

```diff
@@ -70,8 +70,8 @@ export class PermissionsRevokeHandler implements MethodHandler {
     const existingRevokesForGrant: PermissionsRevokeMessage[] = await this.messageStore.query(tenant, query) as PermissionsRevokeMessage[];
 
     // Conflict 409 if the grant already has an older revoke
-    const oldestExistingRevoke = await PermissionsRevoke.getOldestRevoke(existingRevokesForGrant);
-    if (oldestExistingRevoke !== undefined && !(await PermissionsRevoke.isCreatedOlder(oldestExistingRevoke, message))) {
+    const oldestExistingRevoke = await Message.getOldestMessage(existingRevokesForGrant);
+    if (oldestExistingRevoke !== undefined && !(await Message.isNewer(oldestExistingRevoke, message))) {
```

Comment:
Sanity: is this supposed to be `isOlder()`? If so, consider adding a test since this was not caught by existing tests.

Reply:
It is supposed to be `isNewer`. We do have a test which failed when I was deduping the code. I made two mistakes that canceled each other out to produce correct behavior. First, I used `isCreatedOlder` when I should have used `isCreatedNewer`. Second, I wrote a bug in `isCreatedOlder` so it was returning whether the first message was newer.

The number of negations is making my head spin. Let me update it one more time so it's less confusing.